gui/randogui.py

```diff
@@ -708,6 +786,255 @@ def launch_tricks_dialog(self):
+        arcPath: str
+        currentPack: str
+
+        match self.current_model_type:
+            case "Player":
+                arcPath = "Player/Alink.arc"
+                currentPack = self.options["selected-player-model-pack"]
+            case "Loftwing":
+                arcPath = "Loftwing/Bird_Link.arc"
+                currentPack = self.options["selected-loftwing-model-pack"]
+
+        self.ui.option_model_pack_select.blockSignals(True)
+        self.ui.option_model_pack_select.clear()
+        self.ui.option_model_pack_select.addItem("Default")
+        for path in CUSTOM_MODELS_PATH.glob(f"*/{arcPath}"):
+            packName = path.parts[-3]
+            if (
+                packName == "Default"
+            ):  # ignore packs called default so they don't clash with default link
+                continue
+            self.ui.option_model_pack_select.addItem(packName)
+            if packName == currentPack:
+                self.ui.option_model_pack_select.setCurrentText(packName)
+        self.ui.option_model_pack_select.blockSignals(False)
+        self.change_model_pack()
+
+    def change_model_pack(self, index: int = 0):
+        if self.ui.option_model_pack_select.count() < 1:
+            return
+        self.current_model_pack = self.ui.option_model_pack_select.currentText()
+
+        self.read_metadata()
+
+        match self.current_model_type:
+            case "Player":
+                self.options.set_option(
+                    "selected-player-model-pack", self.current_model_pack
+                )
+                if allowTunicSwap := self.metadata.get("AllowTunicSwap"):
+                    if allowTunicSwap == "True":
+                        self.ui.option_tunic_swap.setEnabled(True)
+                    else:
+                        self.ui.option_tunic_swap.setChecked(False)
+                        self.ui.option_tunic_swap.setEnabled(False)
+                        self.options.set_option("tunic-swap", False)
+                else:
+                    self.ui.option_tunic_swap.setEnabled(True)
+
+                self.save_settings()
+            case "Loftwing":
+                self.options.set_option(
+                    "selected-loftwing-model-pack", self.current_model_pack
+                )
+                self.save_settings()
+
+        self.update_model_customisation()
+
+    def read_metadata(self):
+        if self.current_model_pack == "Default":
+            self.metadata_path = (
+                CUSTOM_MODELS_PATH
+                / "Default"
+                / self.current_model_type
+                / "metadata.json"
+            )
+            if not os.path.isfile(self.metadata_path):
+                self.metadata_path.write_bytes(
+                    (
+                        LINK_MODEL_DATA_PATH
+                        / self.current_model_type
+                        / "defaultMetadata.json"
+                    ).read_bytes()
+                )
```

Comment:
```suggestion
            self.metadata_path = (
                CUSTOM_MODELS_PATH
                / "Default"
                / self.current_model_type
                / "metadata.json"
            )
            default_metadata_path = (
                LINK_MODEL_DATA_PATH
                / self.current_model_type
                / "defaultMetadata.json"
             )
             with open(default_metadata_path) as f:
                 default_metadata =json.load(f)
                 self.metadata = default_metadata
             if self.metadata_path.is_file():
                 with open(self.metadata_path) as f:
                     self.metadata = json.load(f)
                     self.metadata['Color'] = default_metadata['Color'] | self.metadata['Color']
```
Basically, the idea is to always read the default metadata and apply the potentially changed metadata on top of it. Because I think otherwise we run into problems, when we have to update the defaultMetadata and have to add something to it.

Reading through the rest of the PR, I had to add a bit of stupid stuff to make merging the colors work.